### ISSUE TITLE
Possible improvements for http api

### ISSUE THREAD

**Issue by marklagendijk** [loadimpact/k6#436]
The current http api is not yet very user friendly. For example: if you want to post JSON data, you manually need to do the serialization and set the `Content-Type` header.
I think this api would be greatly improved, if it would implement certain concepts from the [Node.js request library](https://github.com/request/request#requestoptions-callback).

- Because of method signature limitations of Go (no default params etc), I would make all the methods (`get`, `post`, `request` etc) take a single params object as parameter.
- `baseUrl`
- `qs` - object containing querystring values to be appended to the url
- `body` - entity body for PATCH, POST and PUT requests.
- `form` - when passed an object or a querystring, this sets body to a querystring representation of value, and adds Content-type: application/x-www-form-urlencoded header.
- `json` - sets body to JSON representation of value and adds Content-type: application/json header. Additionally, parses the response body as JSON.

There are more things that could be added, but I consider these as the most important. It is important to be able to easily do JSON and form requests. 
We should try to prevent that people need to do such basic things themselves.

**Comment by liclac** [comment]
I like the idea of taking a page from their book. We could easily change it so that `url` can be either a URL or an object.

**Comment by robingustafsson** [comment]
I like this as well, but I'm not sure if auto parsing the response as JSON if `json` is set is necessary. I mean, isn't this covered by the current [Response API](https://docs.k6.io/docs/response-k6http) with the `json()` method (as I guess the alternative would be to return the same when `body` is accessed or `json()` are called)?

**Comment by marklagendijk** [comment]
@robingustafsson good point. When I wrote this I hadn't yet found out about `response.html()` and `response.json()`. I think we should add some examples about those (the OO practice of having methods on returned data, is less common in the Javascript world).
